### PR TITLE
Feature/universal cart

### PR DESCRIPTION
**Pull request by stojankz**
Openpos-framework changes necessary to implement universal cart

### Issues Fixed
https://github.com/JumpMind/commerce/issues/7593

### Screenshots
universal-cart

### CODE REVIEW

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/universal-cart-item/_universal-cart-item-theme.scss [hunk 1]
```diff
@@ -0,0 +1,20 @@
+@mixin universal-cart-item-theme($theme) {
+    $app-primary: mat-color(map-get($theme, primary));
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/universal-cart-item/_universal-cart-item-theme.scss:2]: Are all of these variables necessary when you only use success and warn?

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/universal-cart-item/universal-cart-item.component.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<div responsive-class class="outer">
+    <div class="item-content">
+        <app-image *ngIf="item.imageUrl" responsive-class class="item-card-image" [imageUrl]="item.imageUrl | imageUrl"
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/universal-cart-item/universal-cart-item.component.html:3]: the imageUrl pipe should be built into the image component so you don't need to run it here

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/universal-cart-item/universal-cart-item.component.html [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            <div class="description-and-availability" responsive-class>
+                <div *ngIf="item.description" responsive-class class="item-card-title">
+                    {{item.description}}
+                </div>
+                <span class="availability" [ngClass]="getAvailabilityStyle()" responsive-class>
+                    {{item.availabilityLabel}}
+                </span>
+            </div>
+            <ul responsive-class class="item-card-details" *ngIf="item.additionalLabels">
+                <li *ngFor="let additionalLabel of item.additionalLabels" class="muted-color" style="margin-top: 4px">
```
Comment (jchirico8) [inline openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/universal-cart-item/universal-cart-item.component.html:16]: Don't use inline styles

File: openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/universal-cart-item/_universal-cart-item-theme.scss (hunk 1 above)
Comment (stojankz) [reply openpos-client-libs/projects/openpos-client-core-lib/src/lib/shared/components/universal-cart-item/_universal-cart-item-theme.scss:2]: nope thats a result of copy pasta, I can remove them

**Review by jchirico8 — APPROVED** [review]

**Review by jchirico8 — DISMISSED** [review]

**Review by jchirico8 — APPROVED** [review]

**Pull request merged**